common-lib/common/variables/variables.libsonnet

```diff
@@ -59,14 +61,30 @@ local utils = import '../utils.libsonnet';
            + var.datasource.generalOptions.withLabel(prometheusDatasourceLabel)
            + var.datasource.withRegex(''),
        },
+       adHoc:
+         var.adhoc.new('adhoc', 'prometheus', '${' + root.datasources.prometheus.name + '}')
+         + var.adhoc.generalOptions.withLabel('Adhoc filters')
```

Comment:
When we write it out, is it "Adhoc", "AdHoc" or "Ad hoc" 

Because my personal take is "ad hoc" but if we have a formal naming for this feature we should use that, just wanting to clarify :)